### ISSUE TITLE
Problem with parallel executive when done in both vstestconsole.exe and also visual studio by enabling parallel execution mode

### ISSUE THREAD

**Issue by manaswinipolkampally** [Microsoft/EasyRepro#1025]
![IMG_20201119_212142](https://user-images.githubusercontent.com/22270535/99690128-cbb0c300-2aad-11eb-8a5e-54662a76c5aa.jpg)

## Bug Report

**EasyRepro Version**
  - [x] Microsoft Dynamics 365 Online Version 9.1 (9.1.x) (DB 9.1.x) online
  
**UCI or Classic Web**
  - [x] UCI (XrmApp)

**Online or On Premise**
  - [x] Online

**Browser**
  - [x] Chrome

**Describe the bug**  
My main objective is to run the tests in parallel without having the use of azure pipeline.
So studied and followed all the steps provided by the company and implement it. My machine is 32 core hence at the max almost 30 process I can run in parallel. This is working properly because of the use of Ms test adapter. I am able to launch and open browsers simultaneously but the catch over here is that the executive of the tests and it's functionality which needed to be covered is taking place with the current window not on all the browsers which are launched. Hence unable to run the tests in parallel.
We had tried multiple options such as using the concept of threads, async and await but nothing was working as expected.
Can you help us out with the issue of why this happening and why we are unable to execute tests in Parallel ?

**Code to reproduce**  

Place this code in Properties/AssemblyInfo.cs file
[ assembly: Parallelize(Workers =3, Scope = ExecutionScope.MethodLevel]

Place the next piece of code in setting file(settings.runsettings) file which is an XML
RunConfiguration
MsTest>
Parallelize>
Workers> 3/Workers>
Scope> MethodLevel/Scope>
/Parallelize>
/MsTest>

 
 
**Expected behavior**  
Launching the browser based on the workers count and tests that needs to be run.

**Screenshots**  

![IMG_20201119_212142](https://user-images.githubusercontent.com/22270535/99689662-44fbe600-2aad-11eb-871f-db7e1c82dee6.jpg)
![IMG_20201119_212132](https://user-images.githubusercontent.com/22270535/99689689-4decb780-2aad-11eb-82db-30a7c9be588c.jpg)

**Comment by manaswinipolkampally** [comment]
It's been a week and no response from team?

**Comment by Shakevg** [comment]
@manaswinipolkampally, 
I just saw an issue with logging during parallel execution.
Could you provide information:
1) How you create a client instance for each thread
2) Stack trace with error

**Comment by badarmunirr** [comment]
@manaswinipolkampally  this problem isn't with mstest it's with chrome doesn't run scripts in the background. it only executes the on-screen script

**Comment by ewingjm** [comment]
I don't think this is a bug with EasyRepro - we're running in parallel with it fine. The issue here is going to be in how you're instantiating `XrmApp` and `WebClient`. You'll need to share that code.

**Comment by manaswinipolkampally** [comment]
> I don't think this is a bug with EasyRepro - we're running in parallel with it fine. The issue here is going to be in how you're instantiating `XrmApp` and `WebClient`. You'll need to share that cod

**Comment by manaswinipolkampally** [comment]
> I don't think this is a bug with EasyRepro - we're running in parallel with it fine. The issue here is going to be in how you're instantiating `XrmApp` and `WebClient`. You'll need to share that code.

Can you please tell me how ur executing the scripts in parallel that too in the same browser.
Can you provide me the steps that u had followed or any documentation so that I could refer and do the same to execute the tests in parallel
![IMG_20201215_182742](https://user-images.githubusercontent.com/22270535/102217897-40d3b480-3f03-11eb-9c74-39bbe64cb49f.jpg)

**Comment by ewingjm** [comment]
You've got `WebClient` and `XrmApp` as static, meaning there will only ever be a reference to one instance. You need to create an instance of each per test method to run in parallel.

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/static

**Comment by badarmunirr** [comment]
I have WebClient , XrmApp  both public but still not running in parallel 
![image](https://user-images.githubusercontent.com/64267888/102331740-06265680-3fad-11eb-9c3d-806303e0e995.png)
Not sure what I am doing wrong here. it's a simple case

**Comment by ewingjm** [comment]
Are you seeing the tests in-progress at the same time in the Test Explorer window? Or are the test methods themselves running sequentially?

**Comment by badarmunirr** [comment]
test methods are running parallel. 
![image](https://user-images.githubusercontent.com/64267888/102332907-87cab400-3fae-11eb-98d9-e23fe6f69fa2.png)

**Comment by ewingjm** [comment]
And you're only seeing one browser instance? I'm not really sure if so, instantiating an `XrmApp` and `WebClient` instance per test is what I've implemented on [powerapps-specflow-bindings](https://github.com/Capgemini/powerapps-specflow-bindings/blob/master/bindings/src/Capgemini.PowerApps.SpecFlowBindings/PowerAppsStepDefiner.cs).

**Comment by badarmunirr** [comment]
I am seeing 2 browser instances, but the problem is only the on-screen browser is executing the test case and the background browser  is stuck

**Comment by manaswinipolkampally** [comment]
> I am seeing 2 browser instances, but the problem is only the on-screen browser is executing the test case and the background browser is stuck

Yes my issue is also the same

**Comment by ewingjm** [comment]
I've never seen that before. Does the test that instantiated the one in the background fail? Have you tried running in headless mode?

**Comment by manaswinipolkampally** [comment]
> I've never seen that before. Does the test that instantiated the one in the background fail? Have you tried running in headless mode?

Yup tried but no Luck!

**Comment by badarmunirr** [comment]
@ewingjm  your test cases are running parallel on chrome?

**Comment by ewingjm** [comment]
Yes, but I use NUnit rather than MSTest.

**Comment by badarmunirr** [comment]
Tried with Nunit still the same error

**Comment by ewingjm** [comment]
What's the error message?

**Comment by badarmunirr** [comment]
Not an error message. getting the same results I got with Mstest browser is getting stuck in the background

**Comment by ewingjm** [comment]
So does the test run itself actually succeed if it doesn't end in an error message? What happens if you put breakpoints on both test methods and try to debug through?

**Comment by badarmunirr** [comment]
I got it running using Firefox but its not working with chrome . but firefox is slow and keeps breaking

**Comment by manaswinipolkampally** [comment]
Can anyone help me how to solve this issue and make my scripts run parallel in Chrome browser.

**Comment by badarmunirr** [comment]
Sure we can connect

**Comment by manaswinipolkampally** [comment]
> Sure we can connect
Can we have a meeting based on your availability
